### PR TITLE
feat(algo): add verify transaction function

### PR DESCRIPTION
**Pull request by akarath**
This pull request enhances Algorand transaction verification by introducing a comprehensive validation flow and a full suite of unit tests. The main change is the implementation of strict checks for transaction parameters and structure, ensuring that only valid transactions are accepted and errors are handled gracefully.

### Transaction Verification Logic

* Added a new `verifyTransaction` method in `algo.ts` that validates required parameters (`txParams`, `txPrebuild`, and `txHex`) and performs type-specific transaction validation for payment, asset transfer, and key registration transactions.
* Implemented helper methods (`validateBasicTransaction`, `validatePayTransaction`, `validateAssetTransferTransaction`, `validateKeyRegTransaction`) to check transaction fields such as sender, recipient, amount, fee, and key registration parameters.

### Unit Test Coverage

* Added extensive unit tests in `verifyTransaction.ts` to check for missing parameters, mismatched amounts or addresses, multiple recipients, malformed transaction hex, and edge cases (such as zero-amount and close remainder transactions).
* Included tests for different transaction types (payment, asset transfer, key registration) and ensured validation works across network variations and with optional transaction fields like memos.

These changes collectively make transaction validation more robust and reliable for Algorand, reducing the risk of accepting malformed or incorrect transactions.WIN-7595

### CODE REVIEW

File: modules/sdk-coin-algo/src/algo.ts [hunk 1]
```diff
@@ -580,6 +580,122 @@ export class Algo extends BaseCoin {
   }
 
   async verifyTransaction(params: VerifyTransactionOptions): Promise<boolean> {
+    const { txParams, txPrebuild } = params;
```
Comment (arvind-bg) [inline modules/sdk-coin-algo/src/algo.ts:583]: Why not abstract out validations to separate validator classes for reusability?

**Review by arvind-bg — DISMISSED** [review]

File: modules/sdk-coin-algo/src/algo.ts (hunk 1 above)
Comment (akarath) [reply modules/sdk-coin-algo/src/algo.ts:583]: thanks for the suggestion. i can segregate it out while implementing for other tx types

**Review by veetragjain — APPROVED** [review]

**Review by arvind-bg — APPROVED** [review]
